### PR TITLE
Add renderMonthElement to react-dates

### PR DESCRIPTION
**Pull request by VincentDamour**
Please fill in this template.

- [ X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X ] Test the change in your own code. (Compile and run.)
- [ X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X ] Increase the version number in the header if appropriate.
- [ X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Base on [readme](https://github.com/airbnb/react-dates) and [resolved issue](https://github.com/airbnb/react-dates/pull/1106)

### CODE REVIEW

**Review by NathanNZ — APPROVED** [review]
Looks fine to me!

File: types/react-dates/tslint.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "dt-header": false,
         "eofline": false,
         "export-just-namespace": false,
+        "file-name-casing": false,
```
Comment (weswigham) [inline types/react-dates/tslint.json]: Uhhhh, why was this lint rule turned off? Keep this one on. For sure.

Comment (VincentDamour) [reply types/react-dates/tslint.json]: If I enable this, tslint wants me to rename the file react-dates-tests.tsx to reactDatesTests.tsx. Doing so, a typescript test will fail, wanting me to rename the file from reactDatesTests.tsx to react-dates.tests.tsx. It's the only way I found to make every test pass.

Comment (andy-ms) [reply types/react-dates/tslint.json]: npm doesn't use the same casing convention tslint wants us to use, so we've disabled that rule globally.

**Pull request merged**